Source/NSIndexPath+Contiguous.swift

```diff
@@ -51,44 +51,44 @@ extension NSIndexPath {
         return nil
     }
 
-    class func indexPathForIndex(collectionView: UICollectionView, index: Int) -> NSIndexPath? {
+    static func indexPathForIndex(_ collectionView: UICollectionView, index: Int) -> IndexPath? {
         var count = 0
-        let sections = collectionView.numberOfSections()
+        let sections = collectionView.numberOfSections
         for section in 0..<sections {
-            let rows = collectionView.numberOfItemsInSection(section)
+            let rows = collectionView.numberOfItems(inSection: section)
             if index >= count && index < count + rows {
                 let foundRow = index - count
-                return NSIndexPath(forRow: foundRow, inSection: section)
+                return IndexPath(row: foundRow, section: section)
             }
             count += rows
         }
 
         return nil
     }
 
-    func totalRow(collectionView: UICollectionView) -> Int {
+    func totalRow(_ collectionView: UICollectionView) -> Int {
         var count = 0
-        let sections = collectionView.numberOfSections()
+        let sections = collectionView.numberOfSections
         for section in 0..<sections {
-            if section < self.section {
-                let rows = collectionView.numberOfItemsInSection(section)
+            if section < (self as NSIndexPath).section {
+                let rows = collectionView.numberOfItems(inSection: section)
                 count += rows
             }
         }
 
-        return count + self.row
+        return count + (self as NSIndexPath).row
     }
 
-    func compareDirection(indexPath: NSIndexPath) -> Direction {
-        let current = self.row * self.section
-        let coming = indexPath.row * indexPath.section
+    func compareDirection(_ indexPath: IndexPath) -> Direction {
+        let current = (self as NSIndexPath).row * (self as NSIndexPath).section
```

Comment:
Casts and more casts and more casts. Something wrong with migration tool or with Swift 3 as a whole?